Sources/PackageRegistry/RegistryClient.swift

```diff
@@ -426,7 +427,10 @@ public final class RegistryClient {
                             // TODO: Bail if archive contains relative paths or overlapping files
                             archiver.extract(from: downloadPath, to: destinationPath) { result in
                                 defer { try? fileSystem.removeFileTree(downloadPath) }
-                                completion(result.mapError { error in
+                                completion(result.tryMap {
+                                    // strip first level component
+                                    try fileSystem.stripFirstLevel(of: destinationPath)
```

Comment:
@yim-lee we would need to adopt this also in the compatibility test suite and update the fixtures accordingly